jmh-core/src/main/java/org/openjdk/jmh/runner/Runner.java

```diff
@@ -475,14 +474,11 @@ private BenchmarkParams newBenchmarkParams(BenchmarkListEntry benchmark, ActionM
                 benchmark.getJvmArgsPrepend().orElse(Collections.<String>emptyList())));
 
         // We want to be extra lazy when accessing ManagementFactory, because security manager
-        // may prevent us doing so. When JMH upgrades to Java 8, replaces this with proper Optional
-        // and lazy Supplier.
-        Optional<Collection<String>> jvmArgsMid = options.getJvmArgs().orAnother(benchmark.getJvmArgs());
-        if (jvmArgsMid.hasValue()) {
-            jvmArgs.addAll(jvmArgsMid.get());
-        } else {
-            jvmArgs.addAll(ManagementFactory.getRuntimeMXBean().getInputArguments());
-        }
+        // may prevent us doing so.
+        jvmArgs.addAll(options.getJvmArgs().orElseGet(
+                () -> benchmark.getJvmArgs().orElseGet(
+                        () -> ManagementFactory.getRuntimeMXBean().getInputArguments())));
+
```

Comment:
Indentation looks a bit odd to me...

```suggestion
        jvmArgs.addAll(options.getJvmArgs()
                .orElseGet(() -> benchmark.getJvmArgs()
                .orElseGet(() -> ManagementFactory.getRuntimeMXBean().getInputArguments())
        ));

```

Reply:
Fixed in e7547c8.